### DISCUSSION TITLE
Specfying sampling factor with colospace breaks JPEG-in-TIFF?

### DISCUSSION THREAD

**Discussion by cdokolas** [ImageMagick/ImageMagick#7615] [Help] [answered]
### ImageMagick version

7.1.1-38

### Operating system, version and so on

Windows 11

### Description

When converting image to JPEG-in-TIFF, I would like to control the sampling factor as well as the components to encode.

Using the command `magick image.png -compress JPEG -colorspace YCbCr image.tif` produces the correct JPEG-in-TIFF type (i.e. the JPEG in the TIFF encodes the YCbCr components). However, when I add the `-sampling-factor` parameter anywhere after `-compress JPEG` (e.g. `magick image.png -compress JPEG -colorspace YCbCr -sampling-factor 4:2:2 image-422.tif`), the image is broken.

Why is the image broken? How can I control the sampling factor for YCbCr components? (better compression with minimum artifacts)

Note: When generating JPEG files, the `-sampling-factor` parameter works fine.

**Comment by fmw42** [comment]
Does it work if you put the -compress JPG at the end?

**Comment by urban-warrior** [comment, marked as answer]
Thank you for reporting the issue. We have successfully reproduced it and are actively working on a patch to resolve it. You can expect this patch to be merged into the [main GIT branch](https://github.com/ImageMagick/ImageMagick), later today. As part of our commitment to quality, this fix will also be included in the upcoming [beta releases of ImageMagick](https://imagemagick.org/archive/beta/) by tomorrow. Your patience and feedback are greatly appreciated.
